### PR TITLE
Image customCss Implementation

### PR DESCRIPTION
**Pull request by dawinValenzuela**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [x] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**Other information**:
Add customCss implementation

### CODE REVIEW

File: packages/styled-components/src/components/primitives/Image/Image.styled.ts [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
   user-select: none;
+
+  ${(props) => customCss(props)}
 `;
 
-export const StyledImage = styled.img`
+export const StyledImage = styled.img<StyledCssProps>`
   user-select: none;
+
+  ${(props) => customCss(props)}
```
Comment (guilean) [inline packages/styled-components/src/components/primitives/Image/Image.styled.ts:16]: Looks identical except the tag. Maybe we can create a single StyledImage which will render an img tag as default and we can provide as="source" for the source tag.

**Review by guilean — CHANGES_REQUESTED** [review]

**Review by guilean — APPROVED** [review]

**Review by tkanzakic — CHANGES_REQUESTED** [review]

File: packages/styled-components/src/components/primitives/Image/Image.stories.tsx [hunk 2]
```diff
@@ -18,5 +26,6 @@ export const Playground = () => (
       ),
     }}
     onLoad={action('Image loaded')}
+    customCss={select('Custom css', customCss, customCss.none)}
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Image/Image.stories.tsx:29]: use a plain text as in other components

**Review by tkanzakic — CHANGES_REQUESTED** [review]

File: packages/styled-components/src/components/primitives/Image/Image.styled.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
+import { customCss } from 'utils/customCss';
 
-export const StyledSource = styled.source`
+export const StyledImage = styled.img<StyledCssProps>`
   user-select: none;
-`;
 
-export const StyledImage = styled.img`
-  user-select: none;
+  ${(props) => customCss(props)}
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Image/Image.styled.ts:8]: this approach is no longer valid, check this PR https://github.com/cmpsr/composer/pull/187

File: packages/styled-components/src/components/primitives/Image/Image.tsx [hunk 4]
```diff
@@ -1,26 +1,31 @@
...
   onLoad,
 }: Props) => {
   return (
     <picture className={className}>
-      <StyledSource
-        className={imageClassName}
+      <StyledImage
+        as="source"
+        css={customCss}
+        className={cn(imageClassName, { custom: customCss })}
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Image/Image.tsx:19]: not necessary with the new approach

File: packages/styled-components/src/components/primitives/Image/Image.tsx [hunk 5]
```diff
@@ -1,26 +1,31 @@
...
         data-testid={testId}
         type="image/webp"
         srcSet={`${url}?fm=webp`}
       />
       <StyledImage
         data-testid="imageContent"
         onLoad={onLoad}
-        className={imageClassName}
+        css={customCss}
+        className={cn(imageClassName, { custom: customCss })}
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Image/Image.tsx:28]: not necessary with the new approach

**Review by tkanzakic — CHANGES_REQUESTED** [review]

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**